src/composables/auth.ts

```diff
@@ -326,12 +329,16 @@ export const useAuth = createCustomScopedComposable(() => {
     checkUserAuth();
   }
 
+  async function checkPassword(password: string) {
+    const key = await generateEncryptionKey(password, encryptionSalt.value!);
+    await decrypt(key, mnemonicEncrypted.value!);
+  }
+
   /**
    * Check if password provided by user is correct and if true update it to new one.
    */
   async function updatePassword(currentPassword: string, newPassword: string) {
```

Comment:
This function is used in nly one place and with your change the body is only 2 lines. I suggest to remove it and use those 2 lines in the `Secure LoginSettings` directly.